clients/cloud/nodejs/producer.js

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * =============================================================================
+ *
+ * Produce messages to Confluent Cloud
+ * Using the node-rdkafka client for Apache Kafka
+ *
+ * =============================================================================
+ */ 
+
+const Kafka = require('node-rdkafka');
+const { configFromCli } = require('./config');
+
+const ERR_TOPIC_ALREADY_EXISTS = 36;
+
+function ensureTopicExists(config) {
+  const adminClient = Kafka.AdminClient.create({
+    'metadata.broker.list': config['bootstrap.servers'],
+    'sasl.username': config['sasl.username'],
+    'sasl.password': config['sasl.password'],
+    'security.protocol': 'SASL_SSL',
+    'sasl.mechanisms': 'PLAIN'
+  });
+
+  return new Promise((resolve, reject) => {
+    adminClient.createTopic({
+      topic: config.topic,
+      num_partitions: 1,
+      replication_factor: 3
+    }, (err) => {
+      if (!err) {
+        console.log(`Created topic ${config.topic}`);
+        return resolve();
+      }
+
+      if (err.code === ERR_TOPIC_ALREADY_EXISTS) {
+        return resolve();
+      }
+
+      return reject(err);
+    });
+  });
+}
+
+function createProducer(config, onDeliveryReport) {
+  const producer = new Kafka.Producer({
+    'metadata.broker.list': config['bootstrap.servers'],
+    'sasl.username': config['sasl.username'],
+    'sasl.password': config['sasl.password'],
+    'security.protocol': 'SASL_SSL',
+    'sasl.mechanisms': 'PLAIN',
+    'dr_cb': true,
+    'dr_msg_cb': true
```

Comment:
I think you can remove `dr_cb` - from the README docs it seems it's unnecessary given `dr_msg_cb` is specified.

Reply:
I found that this was necessary in order to populate `value` in the delivery report object. We could remove the `value` from the message logged to the user and also remove this property, if that's advisable.